source/slang/check.cpp

```diff
@@ -3516,6 +3519,14 @@ namespace Slang
                 EnsureDecl(m);
             }
 
+            if (aggTypeDeclRef)
+            {
+                for (auto inheritanceDecl : decl->getMembersOfType<InheritanceDecl>())
+                {
+                    checkConformance(aggTypeDeclRef.getDecl(), inheritanceDecl);
```

Comment:
This seems all right for now, but will have problems eventually. If I understand it, we are saying that when an extension adds conformances, we validate them by checking the original type (which should now be extended).

The simple problem is that the only reasonable semantics would be that we should check conformances *after* a first pass where we make sure all extensions have been registered. Otherwise each extension will be checked in a different state, where a different set of extensions have been registered.

The next small gotcha is that if an extension ever defined a method that shadowed one in the original type, then that method should be favored by lookup when checking the extension (including conformances it adds), even if ordinary lookup should favor the method declared by the type itself. This implies that the decl-ref we use as the “root” when checking conformances should be the `extension` (although this might be tricky to manage for now).

The final gotcha is that an extension might only apply to certain specializations of a generic type (e.g., `extension Foo<int>`), so if we check conformances for the unspecialized type (`Foo<T>`) the extension won’t apply, and so its methods won’t be found by lookup.